fe/src/components/Watchlist/Watchlist.tsx

```diff
@@ -0,0 +1,84 @@
+import update from "immutability-helper";
+import { StockItem } from "@pages/WatchlistPage";
+import { memo, useCallback, useState } from "react";
+import styled from "styled-components";
+import { WatchlistItem } from "./WatchlistItem";
+
+export const Watchlist = memo(function Watchlist({
+  stockItems,
+}: {
+  stockItems: StockItem[];
+}) {
+  const watchlistTitles = ["종목명", "현재가", "변동률", "배당금", "업종"];
+  const [stocks, setStocks] = useState<StockItem[]>(stockItems);
+
+  const moveStock = useCallback((dragIndex: number, hoverIndex: number) => {
+    setStocks((prevStocks: StockItem[]) =>
+      update(prevStocks, {
+        $splice: [
+          [dragIndex, 1],
+          [hoverIndex, 0, prevStocks[dragIndex] as StockItem],
+        ],
+      })
+    );
+  }, []);
+
+  const renderStock = useCallback((stock: StockItem, index: number) => {
+    return (
+      <WatchlistItem
+        key={stock.id}
+        id={`${stock.id}`}
+        item={stock}
+        index={index}
+        moveStock={moveStock}
+      />
+    );
+  }, []);
+  return (
+    <StyledWatchlist>
+      <WatchlistHeader>
+        {watchlistTitles.map((title, index) => (
+          <WatchListTitle key={index}>{title}</WatchListTitle>
+        ))}
+      </WatchlistHeader>
+      {stocks.map((stock, i) => renderStock(stock, i))}
```

Comment:
```suggestion
      {stocks.map((stock, i) =>  
         <WatchlistItem
            key={stock.id}
            id={`${stock.id}`}
            item={stock}
            index={index}
            moveStock={moveStock}
          />)}
```

함수로 구현하신 이유가 있을까요?

rednerStock을 재사용하지 않는 다면 이렇게 변경해도 똑같지 않나요?

Reply:
드래그앤드롭 예제로 돌아가는지만 확인했는데 수정해야겠네요